### ISSUE TITLE
Hangs whens using --pos on some videos

### ISSUE THREAD

**Issue by olegantonyan** [popcornmix/omxplayer#474]
Build date: Fri, 06 May 2016 14:07:54 +0000
Version   : 6c90c75 [master]
Raspberry Pi 3, Raspbian 8

There is a problem with at least one video file: https://yadi.sk/i/Y8Vh3ssRt52BT
It works fine if you play it from the beginning or from anything between 00:00:00 and 00:05:19. 
For example:
```bash
omxplayer 008._Seabream_in_Crazy_water_with_Gennaro_Contaldo.mkv # => OK
omxplayer 008._Seabream_in_Crazy_water_with_Gennaro_Contaldo.mkv --pos 00:02:11 # => OK
omxplayer 008._Seabream_in_Crazy_water_with_Gennaro_Contaldo.mkv --pos 00:05:19 # => OK
```
But, if you play it from 00:05:20 or any further position, the video doesn't appear and the player stays in this state forever (i.e. not stopping at the end of the file)
```bash
omxplayer 008._Seabream_in_Crazy_water_with_Gennaro_Contaldo.mkv --pos 00:05:24 # => will show 'seek 00:05:24 / 00:05:38' shortly and that's it
```

00:05:20 is the position where sound disappears, so this might be a cause.

**Comment by jehutting** [comment]
Yes, the no sound after 00:05:20 is the cause of the "freeze".
To start the multimedia clock there has to be a video _and_ an audio frame read.
If you add option ```--aidx -1``` then only the video frame is enough to start the clock.

**Comment by olegantonyan** [comment]
Thank you @jehutting 
It works, but with `--aidx -1` there is no sound at all. Is there a way to skip audio frame if doesn't exist and play it otherwise?

**Comment by jehutting** [comment]
Yes ` ` ` --aidx -1` ` ` switches the audio off during the full play of the movie (kind of audio mute).
I don't know if it is possible to skip audio during the play or after a seek. How should the code determine that there are no audio frames after a certain point of time? Is there an ffmpeg flag that could be used? I don't know...

**Comment by popcornmix** [comment]
No easy solution. omxplayer (on firmware size) has a clock that starts when all its input ports have valid data. The ports will be from video (if ffmpeg says stream contains video) and from audio (if ffmpeg says stream contains audio).

But if the audio stops significantly before the video and you seek after that point the clock never starts.
I think all that can be done is to have some timeout if audio is enabled but packets never appear from ffmpeg.

But really this sounds like working around a broken file.

**Comment by olegantonyan** [comment]
For now I have another very dirty hack because the player must work autonomously 24/7 and videos come from users. I have a pre-processing (convert, volume normalize), but I don't know how to detect this broken file with ffmpeg before uploading it to the player.

**Comment by jehutting** [comment]
Maybe you can use something like [this](https://github.com/popcornmix/omxplayer/issues/437#issuecomment-206030418) for creating a kind of watchdog.

**Comment by olegantonyan** [comment]
Thank you @jehutting . I've go something like this. Though it has downsides and it's not really a solution, but at least it works

**Comment by jehutting** [comment]
Agree it isn't the best solution...
I assume the end conclusion is "it is a broken file and omxplayer (or any other player) isn't capable to solve this"? So no further action from omxplayer and @olegantonyan is going to close this issue?

**Comment by olegantonyan** [comment]
Well, if omxplayer really can't do anything, then yes, we should close the issue.
But for me as for developer this is very strange. Why then omxplayer has no trouble playing video without audio track? How is that different?

**Comment by jehutting** [comment]
Ok...let me rephrase my question.

But first...

Upon start, omxplayer analyses the stream. The stream has video and/or has audio frames. Accordingly these findings the multimedia clock is started. So if the stream has only audio, the multimedia clock is started as soon as an audio frame is read. If the stream has only video, the clock is started as soon as a video frame is read. If the stream has both video and audio frames then the clock is  started after an audio frame _as well as_ a video frame is read. 

After you seek (/position) the stream the clock start condition has to be met in order to start the clock.

Now in case you seek to a point in time where there are no longer audio or video frames, it is possible that the condition to start the clock is not being full filled. Hence the freeze.

To fix this -as popcornmix suggested (and if I understood it correct)- there should be a timeout (for xx msec) after the "clock starts waiting for the first audio/video frame". If the timeout elapses the clock should no longer wait for the missing frame and omxplayer should just start playing the stream.

The question... Are the missing audio frames after the seek beyond 00:05:20 really due to a broken file (if yes, then close the issue), or is it possible that a stream has no longer audio frames but still has video frames (it is a normal/correct stream, and if so, should we built a timeout in omxplayer)?

**Comment by olegantonyan** [comment]
Got it. Thank you @jehutting 

About broken file. The file may be broken, but it doesn't matter for me because files are uploaded by end users. And I don't know how to detect such broken conditions so I cannot tell them "hey, this file is broken, do something". And users usually won't be able to see if the file is broken because starting at 00:05:20 or so is rare condition (5-10 times a day on my device). And the device playing these videos is completely autonomous (controlled by a server).

This is not critical blocker since I have dirty hack, but the hack has own downsides observable by users.

So, timer in omxplayer will help. I probably won't be able to do this because it's quite complicated (and will probably require weeks to understand how it works and create this functionality). 
Another good solution is to detect broken file with ffmpeg or something like this. But I couldn't find a way to do this. I'm pretty sure it's possible.

To summarize. 
This is not critical bug, no one fired or hurt since there is a dirty workaround.
To build or not to build the timer in omxplayer depends on your understanding what is good for the project (thinking as a product manager). Obviously you don't owe me anything.

btw, Thanks for the great project!

**Comment by jehutting** [comment]
Thanks for your clarification.

If ffmpeg would have indicators that the audio or video stream has ended, life would (probably) simple :-)

Your project seems to be great too. Always interesting to see how omxplayer is used.
